File: Python/Mahasiswa.py
# Mahasiswa adalah bagian dari sivitas akademik

from SivitasAkademik import SivitasAkademik

class Mahasiswa(SivitasAkademik):
    def __init__(self, nik="-", nama="-", jenis_kelamin='-', asal_universitas="-", email_edu="-", fakultas="-", nim="-"):
        super().__init__(nik, nama, jenis_kelamin, asal_universitas, email_edu, fakultas)      # memanggil constructor parent nya
        self.__nim = nim

    def setNim(self, nim):
        self.__nim = nim

    def getNim(self):
        return self.__nim
	

File: cpp/Mahasiswa.cpp
// mahasiswa adalah sivitas akademik

class Mahasiswa : public SivitasAkademik{
	private:
		string nim;
	public:
		Mahasiswa(){
			this->nim = "-";
		}
		Mahasiswa(string nik, string nama, string jenis_kelamin, string asal_universitas, string email_edu, string fakultas, string nim)
		: SivitasAkademik(nik, nama, jenis_kelamin, asal_universitas, email_edu, fakultas){
			this->nim = nim;
		}
		void setNim(string nim){
			this->nim = nim;
		}
		string getNim(){
			return this->nim;
		}
};	
